File: leetcode/0001_two-sum/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums : List[int], target: int) -> List[int]:
        dict = {}
        for idx in range(len(nums)):
            other = target - nums[idx]
            if other in dict:
                return [dict[other], idx]
            dict[nums[idx]] = idx
        return None
    
s = Solution()
nums = [2, 7, 11, 15]
target = 18
print(s.twoSum(nums, target))

File: leetcode/0001_two-sum/two_sum.go
package main

import "fmt"

func main() {
	nums := []int{2, 7, 11, 15}
	fmt.Println("下标为", twoSum(nums, 9))
}

func twoSum(nums []int, target int) []int {
	findMap := make(map[int]int)
	for idx, val := range nums {
		other := target - val
		if v, ok := findMap[other]; ok {
			return []int{v, idx}
		}
		findMap[val] = idx
	}
	return nil
}
